### DISCUSSION TITLE
Can I upgrade directly from v18 to v21?

### DISCUSSION THREAD

**Discussion by syntaxerror00100** [keycloak/keycloak#20724] [Q&A]
I'm currently using v18. Can I upgrade directly from v18 to v21 and skip versions v19 and v20?

**Comment by nicolas-goudry** [comment]
I just performed an upgrade from v16 to v21 without any issues so far, so it’s definitely possible… However, I can’t assure you 100% that it will work for your use case.

I’d suggest that you clone your database and try the migration on this database to make sure everything works fine before migrating the real database.

**Reply by chethankumar4046** [reply]
@nicolas-goudry I have two keyclaok instance in my k8s infra 
1) V18 
2 V21 
How to copy all the data (config, realm. users, themes and all other setting like google Oauth) in shot,
And how to enable export and import option in keycloak

**Reply by nicolas-goudry** [reply]
@chethankumar4046 If there are existing realms/users on the v21 instance that prevent you from recreating it from scratch (DB included), you would have to follow [this guide](https://www.keycloak.org/server/importExport) to first export the realm and its users from the v18 instance before importing it on the v21 instance.

As stated in the documentation, the instances need to be stopped when performing the export/import.

---

About the themes, if you're on K8S you have two choices :
- bundle the theme in the container image (either as a JAR provider or static files)
- mount theme files to the Deployment (from whatever source fits your use case: configmap or volume)

The documentation on themes is very complete, and the section about [deploying themes](https://www.keycloak.org/docs/21.1.1/server_development/#deploying-themes) is what you seek.
